src/js/game/LevelMVC/FacingDirection.js

```diff
@@ -1,6 +1,50 @@
-module.exports = Object.freeze({
+const FacingDirection = Object.freeze({
+  North: 1,
+  South: 2,
+  East: 4,
+  West: 8,
+
+  opposite: function (cardinal) {
+    switch (cardinal) {
+      case FacingDirection.North: return FacingDirection.South;
+      case FacingDirection.South: return FacingDirection.North;
+      case FacingDirection.East: return FacingDirection.West;
+      case FacingDirection.West: return FacingDirection.East;
+    }
+  },
+
+  turnDirection: function (from, to) {
+    switch (from) {
+      case FacingDirection.North: return to === FacingDirection.East ? 'right' : 'left';
+      case FacingDirection.South: return to === FacingDirection.West ? 'right' : 'left';
+      case FacingDirection.East: return to === FacingDirection.South ? 'right' : 'left';
+      case FacingDirection.West: return to === FacingDirection.North ? 'right' : 'left';
+    }
+  },
+
+  // TODO: unify with N/S/E/W above.
```

Comment:
I made a huge mistake by adding North/South/East/West instead of re-using Up/Down/Right/Left. 😦

Reply:
Having both concepts is definitely confusing. How much work will it be to unify them?

Reply:
It shouldn't be too bad — I definitely want to do it before merging this PR.